packages/main/src/StepInput.js

```diff
@@ -0,0 +1,651 @@
+		/**
+		 * Defines the value state message that will be displayed as pop up under the <code>ui5-step-input</code>.
+		 * <br><br>
+		 *
+		 * <b>Note:</b> If not specified, a default text (in the respective language) will be displayed.
+		 * <br>
+		 * <b>Note:</b> The <code>valueStateMessage</code> would be displayed,
+		 * when the <code>ui5-step-input</code> is in <code>Information</code>, <code>Warning</code> or <code>Error</code> value state.
+		 * @type {HTMLElement}
+		 * @slot
+		 * @public
+		 */
+		valueStateMessage: {
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.StepInput.prototype */ {
+		/**
+		 * Fired when the input operation has finished by pressing Enter or on focusout.
+		 *
+		 * @event
+		 * @public
+		*/
+		change: {},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-step-input</code> consists of an input field and buttons with icons to increase/decrease the value
+ * with the predefined step.
+ *
+ * The user can change the value of the component by pressing the increase/decrease buttons,
+ * by typing a number directly, by using the keyboard up/down and page up/down,
+ * or by using the mouse scroll wheel. Decimal values are supported.
+ *
+ * <h3>Usage</h3>
+ *
+ * The default step is 1 but the app developer can set a different one.
+ *
+ * App developers can set a maximum and minimum value for the <code>StepInput</code>.
+ * The increase/decrease button and the up/down keyboard navigation become disabled when
+ * the value reaches the max/min or a new value is entered from the input which is greater/less than the max/min.
+ *
+ * <i>When to use</i>
+ * <ul>
+ * <li>To adjust amounts, quantities, or other values quickly.</li>
+ * <li>To adjust values for a specific step.</li>
+ * </ul>
+ *
+ * <i>When not to use</i>
+ * <ul>
+ * <li>To enter a static number (for example, postal code, phone number, or ID). In this case,
+ * use the regular <code>ui5-input</code> instead.</li>
+ * <li>To display a value that rarely needs to be adjusted and does not pertain to a particular step.
+ * In this case, use the regular <code>ui5-input</code> instead.</li>
+ * <li>To enter dates and times. In this case, use date/time related components instead.</li>
+ * </ul>
+ *
+ * For the <code>ui5-step-input</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/StepInput.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.StepInput
+ * @extends UI5Element
+ * @tagname ui5-step-input
+ * @since 1.0.0-rc.12
+ * @public
+ */
+class StepInput extends UI5Element {
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+
+		// spin variables
+		this.INITIAL_WAIT_TIMEOUT = 500; // milliseconds
+		this.ACCELERATION = 0.8;
+		this.MIN_WAIT_TIMEOUT = 50; // milliseconds
+		this.INITIAL_SPEED = 120; // milliseconds
+
+		if (this._previousValue === undefined) {
```

Comment:
You can't access and manipulate the metadata in the `constructor`. This should be in `onBeforeRendering`

Reply:
Done